pkg/apis/devices.harvesterhci.io/v1beta1/pcidevice.go

```diff
@@ -108,15 +110,18 @@ func resourceName(dev *pci.Device) string {
 	return fmt.Sprintf("%s/%s", vendorCleaned, dev.Product.ID)
 }
 
-func (status *PCIDeviceStatus) Update(dev *pci.Device, hostname string) {
+func (status *PCIDeviceStatus) Update(dev *pci.Device, hostname string, iommuGroups map[string]int) {
 	status.Address = dev.Address
 	status.VendorId = dev.Vendor.ID
 	status.DeviceId = dev.Product.ID
 	status.ClassId = fmt.Sprintf("%s%s", dev.Class.ID, dev.Subclass.ID)
 	// Generate the ResourceName field, this is used by KubeVirt to schedule the VM to the node
 	status.ResourceName = resourceName(dev)
 	status.Description = description(dev)
-
+	group, ok := iommuGroups[dev.Address]
+	if ok {
+		status.IOMMUGroup = strconv.Itoa(group)
+	}
```

Comment:
```suggestion
	if iommuGroup, err := device_manager.Handler.GetDeviceIOMMUGroup(pciBasePath, dev.Address); err == nil {
		status.IOMMUGroup = iommuGroup
	}
```

Reply:
```go
import device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"

const pciBasePath = "/sys/bus/pci/devices"
```

Reply:
I suggest switching to KubeVirt's `DeviceHandler` and drop our `iommu` package entirely.